src/main/java/appeng/decorative/solid/BlockChargedQuartzOre.java

```diff
@@ -49,27 +48,20 @@ public BlockChargedQuartzOre()
 	@Override
 	public Item getItemDropped( final IBlockState state, final Random rand, final int fortune )
 	{
-		for( final Item charged : AEApi.instance().definitions().materials().certusQuartzCrystalCharged().maybeItem().asSet() )
-		{
-			return charged;
-		}
-
-		throw new MissingDefinition( "Tried to access charged certus quartz crystal, even though they are disabled" );
+		return AEApi.instance().definitions().materials().certusQuartzCrystalCharged().maybeItem()
+				.orElseThrow( () -> new MissingDefinition( "Tried to access charged certus quartz crystal, even though they are disabled" ) );
 	}
 
 	@Override
 	public int damageDropped( final IBlockState state )
 	{
-		for( final ItemStack crystalStack : AEApi.instance().definitions().materials().certusQuartzCrystalCharged().maybeStack( 1 ).asSet() )
-		{
-			return crystalStack.getItemDamage();
-		}
-
-		throw new MissingDefinition( "Tried to access charged certus quartz crystal, even though they are disabled" );
+		return AEApi.instance().definitions().materials().certusQuartzCrystalCharged().maybeStack( 1 )
+				.orElseThrow( () -> new MissingDefinition( "Tried to access charged certus quartz crystal, even though they are disabled" ) )
+				.getItemDamage();
```

Comment:
Interesting. I have not consider that option. I used `return maybeItemStack.map( ItemStack::getItemDamage ).orElseThrow()` here. Probably because I personally try to avoid exceptional behaviour/handling during a chain and but it at the end.